### DISCUSSION TITLE
Send vendor Information in Discovery Request

### DISCUSSION THREAD

**Discussion by tlifschitz** [openthread/openthread#8659] [Q&A] [answered]
Hi!

Has anyone tried and succeeded to use `otThreadSetJoinerAdvertisement` API to append custom vendor data (eg a serial number) to a Discovery Request?

I think this is not implemented on the FTD side, since `MeshCoP::JoinerAdvertisementTlv` is not parsed when receiving the Discovery Request, and also there is no CLI command to retrieve this information.

Can anyone confirm if this is not implemented? In that case what would be the easiest way to send custom data during joining process?

Note: 
- [Here](https://github.com/openthread/openthread/blob/main/src/core/thread/discover_scanner.cpp#L115-L134) `DiscoverScanner::Discover` appends advertising data to the discovery request message.
- [Here](https://github.com/openthread/openthread/blob/main/src/core/thread/mle_router.cpp#L2678-L2694) `MleRouter::HandleDiscoveryRequest` parses the tlvs in the discovery request message.

**Comment by tlifschitz** [comment]
@jwhui maybe you can you shed some light on this. Thanks

**Comment by jwhui** [comment, marked as answer]
The Joiner Advertisement feature was something that we were prototyping, but never progressed. Note that it is not currently a part of the Thread Specification.

Some related PRs that were never merged:
- https://github.com/openthread/openthread/pull/5381
- https://github.com/openthread/openthread/pull/5406

**Reply by tlifschitz** [reply]
Thanks for the pointer Jonathan! Is there any specific reason for that not having been merged? Do you think however it is a useful feature for the general use case?

If so, and considering today it is not part of the spec, would you consider merging a PR with its implementation using a conditional compile switch (turned off by default)? Otherwise we can manage to send this custom data from the joiner callback.

**Reply by jwhui** [reply]
We didn't merge because it's not a standard Thread feature and we decided not to move forward with our intended use case.

We can consider merging this feature with a build flag that is disabled by default.

If you find it useful, we should also consider standardizing it within Thread Group.
